### PR TITLE
log: not rotate log file when startup

### PR DESCRIPTION
**Pull request by lowzj**
### Ⅱ. Does this pull request fix one issue?

fixes part of : #921

### CODE REVIEW

File: dfget/core/downloader/p2p_downloader/p2p_downloader.go [hunk 1]
```diff
@@ -188,13 +188,14 @@ func (p2p *P2PDownloader) Run() error {
...
 				p2p.finishTask(response, clientWriter)
 				return nil
-			} else if code == constants.CodePeerWait {
-				continue
-			}
-
-			logrus.Warnf("request piece result:%v", response)
-			if code == constants.CodeSourceError {
-				p2p.cfg.BackSourceReason = config.BackSourceReasonSourceError
+			} else {
```
Comment (Starnop) [inline dfget/core/downloader/p2p_downloader/p2p_downloader.go]: Redundant `else`.

File: pkg/dflog/log.go [hunk 2]
```diff
@@ -61,7 +61,7 @@ func WithLogFile(f string) Option {
 		if logger := getLumberjack(l); logger == nil {
 			l.SetOutput(&lumberjack.Logger{
 				Filename:   f,
-				MaxSize:    20, // mb
+				MaxSize:    40, // mb
```
Comment (Starnop) [inline pkg/dflog/log.go]: make it configurable? WDYT? And so do with the `MaxBackups`.

**Comment by yeya24** [comment]
What will happen if the log file exceed the max file size 40MB?

File: dfget/core/downloader/p2p_downloader/p2p_downloader.go (hunk 1 above)
Comment (lowzj) [reply dfget/core/downloader/p2p_downloader/p2p_downloader.go]: It's not redundant. It makes the `warn log` only outputted when getting an unexpected response code.

File: pkg/dflog/log.go (hunk 2 above)
Comment (lowzj) [reply pkg/dflog/log.go]: agree. There's already a function named `WithMaxSizeMB` to configure `MaxSize`. It's better to open a new pr to implement this especially make the supernode's log configurable.

File: dfget/core/downloader/p2p_downloader/p2p_downloader.go (hunk 1 above)
Comment (Starnop) [reply dfget/core/downloader/p2p_downloader/p2p_downloader.go]: Yes, you are right. Sorry for not have a look at the code carefully.

**Comment by Starnop** [comment]
LGTM.

**Comment by Starnop** [comment]
> What will happen if the log file exceed the max file size 40MB?

FYI. https://github.com/natefinch/lumberjack

image

**Pull request merged**